agent/consul/leader_connect_ca.go

```diff
@@ -909,6 +909,9 @@ func (c *CAManager) UpdateConfiguration(args *structs.CARequest) (reterr error)
 		if respErr, ok := resp.(error); ok {
 			return respErr
 		}
+		if respOk, ok := resp.(bool); ok && !respOk {
+			return errors.New("configuration not applied")
```

Comment:
Perhaps we could make this more descriptive so operators understand why not?
```suggestion
			return errors.New("CA configuration not applied as the given ModifyIndex is older than the current one")
```

Reply:
I wonder how this will affect the HTTP API - did you try it? I assume this error just gets returned to the user.

I notice that we don't document the Check-and-set behaviour at all in this specific API endpoint anyway (maybe another thing to fix) https://www.consul.io/api-docs/connect/ca#update-ca-configuration.

Could use something similar to the words in Config Entries: https://www.consul.io/api-docs/config#cas

Looking though all other endpoints that have optional CAS semantics do so with an explicit field like `cas` rather than just implicitly on the ModifyIndex. I wonder why we choose to do that in this case and then not document it?

Perhaps we should make it explicit instead and ignore `ModifyIndex` entirely which is probably what operators assume we would do and the same as other similar endpoints.

Reply:
I'm not sure that the error should be specific to the `ModifyIndex` in here, by returning an error in connect_ca.go this will not be triggered and act as a fail safe/generic error in case another case happen where we return a nil error with a false. My understanding is in this case the false response mean that we did not modify the config and need to be associated with an error that state the reason, this is the case that we did not write the config but for no reason, which should never happen.

I agree on being explicit it will make it clear and easier to document. 
Is there any concern with retro compatibility if we decide to add a `cas` field?

Reply:
I ended up removing this, I opened #10660 to refactor and remove the bool response